### PR TITLE
Embed Youtube videos in posts for valid Youtube URLs

### PR DESCRIPTION
**Pull request by GxqoRR**
For issue #437 

Will change both editing box preview and posted content:

image
image

### CODE REVIEW

**Comment by ekzyis** [comment]
> First-time contributor

:eyes: Welcome!

I'll try to take a look at this today

File: components/text.js [hunk 1]
```diff
@@ -129,7 +145,14 @@ export default memo(function Text ({ nofollow, imgproxyUrls, children, tab, ...o
             // we don't render it as an image since it was probably a concious choice to include text.
             const text = children[0]
             if (!!text && !/^https?:\/\//.test(text)) {
-              return <a target='_blank' rel={`noreferrer ${nofollow ? 'nofollow' : ''} noopener`} href={href}>{text}</a>
+              // Render video if it's a valid Youtube url, otherwise null
+              const video = renderYoutubeUrl(href);
+              return (
+                <>
+                  <a target='_blank' rel={`noreferrer ${nofollow ? 'nofollow' : ''} noopener`} href={href}>{text}</a>
+                  {video}
```
Comment (ekzyis) [inline components/text.js:153]: Won't this show the link _and_ the embedded video player?

**Comment by huumn** [comment]
Nice work! Welcome to the team. Please share a lightning address so I can send you some sats.

File: components/text.js (hunk 1 above)
Comment (ekzyis) [reply components/text.js:157]: > Btw do you know if the hot reload is working? Seems like I'll have to restart docker container every time after I made changes to see the update, and it takes a couple of minutes. Thanks!

Mhh, it should work for the app but I am using `npm run dev` now. When I was using docker, hot reload worked.

Comment (GxqoRR) [reply components/text.js:157]: No worries. I think huumn has taken care of it. And thanks for the info

**Comment by GxqoRR** [comment]
@huumn Thanks! Glad I could help. Here's my Stacker News lightning address: GxqoRR@stacker.news

**Pull request merged**